test/gtest/uct/test_sockaddr.cc

```diff
@@ -1043,6 +1049,85 @@ UCS_TEST_P(test_uct_sockaddr, ep_disconnect_err_codes)
 UCT_INSTANTIATE_SOCKADDR_TEST_CASE(test_uct_sockaddr)
 
 
+class test_uct_sockaddr_listener : public test_uct_sockaddr {
+protected:
+    test_uct_sockaddr_listener()
+    {
+        m_close_listener_notify = false;
+        m_server_uct_listener   = nullptr;
+    }
+
+    static void conn_request_cb(
+            uct_listener_h listener, void *arg,
+            const uct_cm_listener_conn_request_args_t *conn_req_args)
+    {
+        test_uct_sockaddr_listener *self =
+                static_cast<test_uct_sockaddr_listener*>(arg);
+
+        self->handle_conn_request(listener, conn_req_args,
+                                  test_uct_sockaddr::server_connect_cb,
+                                  test_uct_sockaddr::server_disconnect_cb);
+
+        ASSERT_EQ(nullptr, self->m_server_uct_listener);
+
+        // Notify listener_close_thread to destroy the listener
+        self->m_close_listener_notify = true;
+
+        // Wait until uct_listener_destroy is being called from
+        // listener_close_thread and dereference the server's listener to check
+        // whether the object is valid or not
+        while (self->m_server_uct_listener == nullptr) {
+            std::this_thread::yield();
+        }
+
+        ASSERT_TRUE(self->m_server_uct_listener.load()->cm != nullptr);
+    }
+
+    static void listener_close_thread(test_uct_sockaddr_listener *self)
+    {
+        while (!self->m_close_listener_notify) {
+            std::this_thread::yield();
+        }
+
+        self->m_server_uct_listener = self->m_server->revoke_listener();
+        uct_listener_destroy(self->m_server_uct_listener);
+    }
+
+protected:
+    std::atomic<bool>           m_close_listener_notify;
+    std::atomic<uct_listener_h> m_server_uct_listener;
```

Comment:
```suggestion
    std::atomic<bool>           m_close_listener_notify{false};
    std::atomic<uct_listener_h> m_server_uct_listener{nullptr};
```

If both these values are value-initialized, then compiler-generated default ctor will have the same behavior as the ctor defined above. So manual-defined ctor can be removed in this case.

Reply:
if not initialize `std::atomic`'s, they contain garbage